### PR TITLE
try to solve android arch

### LINKED ISSUE #41: panic at runtime on android
```
E/Go: panic: unaligned 64-bit atomic operation
E/Go: 
E/Go: goroutine 8 [running]:
E/Go: runtime/internal/atomic.panicUnaligned()
E/Go: 	/Users/milerius/sdk/go1.16.2/src/runtime/internal/atomic/unaligned.go:8 +0x38
E/Go: runtime/internal/atomic.Store64(0x8684e2f4, 0x4c4b40, 0x0)
E/Go: 	/Users/milerius/sdk/go1.16.2/src/runtime/internal/atomic/asm_386.s:213 +0x10
E/Go: github.com/kpango/fastime.(*Fastime).StartTimerD.func1(0x8684e2c0, 0x4c4b40, 0x0, 0xb7422b9c, 0x86812820)
E/Go: 	/var/folders/9w/h3xyrgx538g23bw10m59tbgc0000gn/T/gomobile-work-681244108/pkg/mod/github.com/kpango/fastime@v1.0.16/fastime.go:188 +0x88
E/Go: created by github.com/kpango/fastime.(*Fastime).StartTimerD
E/Go: 	/var/folders/9w/h3xyrgx538g23bw10m59tbgc0000gn/T/gomobile-work-681244108/pkg/mod/github.com/kpango/fastime@v1.0.16/fastime.go:185 +0xc9
A/libc: Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 8080 (Thread-2), pid 8045 (e.myapplication)
```

https://github.com/golang/go/issues/11891

Would be nice if we can solve it, i try to make fasthttp work on my android device, and i'm block at this point

cc @kpango i dont know if you ever see this problem

> Reference: [golang/go#599](https://github.com/golang/go/issues/599) and https://golang.org/pkg/sync/atomic/#pkg-note-BUG
> Solution: [golang/go#11891 (comment)](https://github.com/golang/go/issues/11891#issuecomment-433623786)

### PR DESCRIPTION
**Pull request by Milerius**
cc @kpango trying to fix #41  #31

### CODE REVIEW

**Comment by mratsim** [comment]
Doesn't go have something similar to C++ alignas? https://en.cppreference.com/w/cpp/language/alignas

I'm surprised Go atomics aren't aligned automatically to the underlying size.

Assuming Go atomatically aligns trivial types to their size, yes this will ensure 64-bit alignment.

However from a performance point of view, you might want to separate atomics by about a cacheline (64-byte) for those accessed by multiple threads to avoid false sharing.

**Comment by Milerius** [comment]
> Doesn't go have something similar to C++ alignas? https://en.cppreference.com/w/cpp/language/alignas
> 
> I'm surprised Go atomics aren't aligned automatically to the underlying size.
> 
> Assuming Go atomatically aligns trivial types to their size, yes this will ensure 64-bit alignment.
> 
> However from a performance point of view, you might want to separate atomics by about a cacheline (64-byte) for those accessed by multiple threads to avoid false sharing.

https://golang.org/pkg/sync/atomic/#pkg-note-BUG it seems that's is a known bug

**Comment by kpango** [comment]
@Milerius Thank you for the PR. I would love to incorporate this fix if it solves the long standing problem .
Have you tested this on an Android device?
I don't have an Android device on hand, so I can't test it.

File: fastime.go [hunk 1]
```diff
@@ -11,17 +11,18 @@ import (
...
 	uut           uint32
 	uunt          uint32
+	running       *atomic.Value
+	t             *atomic.Value
 	ft            *atomic.Value
 	format        *atomic.Value
 	cancel        context.CancelFunc
 	correctionDur time.Duration
-	dur           int64
+
```
Comment (kpango) [inline fastime.go:25]: time.Duration is also int64 type I think it's better to write like this
```
	uut           uint32
	uunt          uint32
	ut            int64
	unt           int64
	dur           int64
	correctionDur time.Duration
	running       *atomic.Value
	t             *atomic.Value
	ft            *atomic.Value
	format        *atomic.Value
	cancel        context.CancelFunc
```

**Comment by Milerius** [comment]
> @Milerius Thank you for the PR. I would love to incorporate this fix if it solves the long standing problem .
> Have you tested this on an Android device?
> I don't have an Android device on hand, so I can't test it.

Hmm I would like an advice how to use this patched version of glg in my program to test my fix also https://github.com/kpango/glg/issues/107 seems to break android compatibility, good things is you can test this local fix by compiling for ARM and use qemu to launch example, but for android i need a way to build glg with my patch

**Comment by Milerius** [comment]
Something you can try @kpango is:

install android studio.

Compile your module with `gomobile bind -v --target=android .`

and import it in android studio as mentionned in: https://medium.com/@LogPacker/gomobile-library-development-for-ios-android-8711e76817d8

You do not need any android device because there is a simulator

**Comment by kpango** [comment]
Thanks for providing the verification method, I guess I need to release it for Dependency to try it on Android, I'll keep the Issue and merge this PR and release Patch for glg as well.
I'll contact you about it, can you verify it for me?

**Pull request merged**